### PR TITLE
Enhancement/#9654 - Integrate First-Party Mode Toggle into Analytics Settings

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #9654 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/settings/SettingsForm.stories.js [hunk 1]
```diff
@@ -24,6 +29,7 @@ import { Cell, Grid, Row } from '../../../../material-components';
 import { MODULES_ANALYTICS_4 } from '../../datastore/constants';
 import { provideModules } from '../../../../../../tests/js/utils';
 import WithRegistrySetup from '../../../../../../tests/js/WithRegistrySetup';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/settings/SettingsForm.stories.js]: Updated ✅

File: assets/js/modules/analytics-4/components/settings/SettingsForm.js [hunk 2]
```diff
@@ -35,15 +35,19 @@ import { TrackingExclusionSwitches } from '../common';
...
-import EntityOwnershipChangeNotice from '../../../../components/settings/EntityOwnershipChangeNotice';
-import { isValidAccountID } from '../../utils/validation';
 import ConversionTrackingToggle from '../../../../components/conversion-tracking/ConversionTrackingToggle';
-import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import EntityOwnershipChangeNotice from '../../../../components/settings/EntityOwnershipChangeNotice';
+import FirstPartyModeToggle from '../../../../components/first-party-mode/FirstPartyModeToggle';
 import Link from '../../../../components/Link';
 import SettingsGroup from '../../../../components/settings/SettingsGroup';
+import { isValidAccountID } from '../../utils/validation';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/settings/SettingsForm.js]: Updated ✅

File: assets/js/modules/analytics-4/components/settings/SettingsForm.stories.js [hunk 3]
```diff
@@ -92,6 +98,78 @@ EnhancedMeasurementSwitch.decorators = [
...
+					isFPMHealthy: false,
+					isScriptAccessEnabled: false,
+				},
+			} );
+
+			registry.dispatch( CORE_SITE ).receiveGetFirstPartyModeSettings( {
+				isEnabled: true,
+				isFPMHealthy: false,
+				isScriptAccessEnabled: false,
+			} );
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/settings/SettingsForm.stories.js]: Updated ✅

File: assets/js/modules/analytics-4/components/settings/SettingsForm.stories.js [hunk 4]
```diff
@@ -92,6 +98,78 @@ EnhancedMeasurementSwitch.decorators = [
...
+					isFPMHealthy: true,
+					isScriptAccessEnabled: true,
+				},
+			} );
+
+			registry.dispatch( CORE_SITE ).receiveGetFirstPartyModeSettings( {
+				isEnabled: true,
+				isFPMHealthy: true,
+				isScriptAccessEnabled: true,
+			} );
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/settings/SettingsForm.stories.js]: Updated ✅

File: assets/js/modules/analytics-4/components/settings/SettingsForm.stories.js [hunk 5]
```diff
@@ -92,6 +98,78 @@ EnhancedMeasurementSwitch.decorators = [
...
+		return (
+			<WithRegistrySetup func={ setupRegistry }>
+				<Story />
+			</WithRegistrySetup>
+		);
+	},
+];
+WithFirstPartyModeEnabled.scenario = {};
+
+export const WithFirstPartyModeDisabled = Template.bind( null );
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/settings/SettingsForm.stories.js]: Updated ✅

**Review by techanvil — APPROVED** [review]
LGTM, nice one @hussain-t!

**Pull request merged**